File: Languages/MATLAB/serial_sum.m
%===========================================================================
% Program: serial_sum( N )
%          Calculates integer sum from 1 to N
%
% Run:     matlab -nodesktop -nodisplay -nosplash -r "serial_sum(100); exit"
%===========================================================================
function s = serial_sum(N) 
  s = 0; 
  for i = 1:N 
    s = s + i; 
  end 
  fprintf('Sum of integers from 1 to %d is %d.\n', N, s); 
end

File: Parallel_Computing/EP/Example2/serial_sum.R
#================================================
# Function: serial_sum(N)
#           Returns the sum of integers from 1
#           through N
#================================================
serial_sum <- function(x){
  k <- 0
  s <- 0
  while (k < x){
    k <- k + 1
    s <- s + k
  }
  return(s)
}

# +++ Main program +++
x <- as.integer(Sys.getenv('inp'))
res <- serial_sum(x)
print(res)
